lib/scout_apm/auto_instrument/rails.rb

```diff
@@ -0,0 +1,73 @@
+
+# In order for this to work, you must add `gem 'parser'` to your Gemfile.
+require 'parser/current'
+
+module ScoutApm
+  module AutoInstrument
+    module Rails
+      def self.rewrite(path, code = nil)
+        code ||= File.read(path)
+
+        ast = Parser::CurrentRuby.parse(code)
+
+        # pp ast
+
+        buffer = Parser::Source::Buffer.new(path)
+        buffer.source = code
+
+        rewriter = Rewriter.new
+
+        # Rewrite the AST, returns a String with the new form.
+        rewriter.rewrite(buffer, ast)
+      end
+
+      class Rewriter < ::Parser::TreeRewriter
```

Comment:
Going to need you to walk me through the logic used in this class just so I understand fully :)